extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/spi/datafetcher/BlockingHelper.java

```diff
@@ -1,14 +1,19 @@
 package io.quarkus.smallrye.graphql.runtime.spi.datafetcher;
 
 import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
 
+import io.quarkus.arc.Arc;
+import io.quarkus.virtual.threads.VirtualThreads;
 import io.smallrye.graphql.schema.model.Execute;
 import io.smallrye.graphql.schema.model.Operation;
 import io.vertx.core.Context;
 import io.vertx.core.Promise;
 
 public final class BlockingHelper {
 
+    private static Boolean isVirtualThreadsExecutorAvailable = null; //todo or no need to cache this?
```

Comment:
The executor is application-scoped, so I guess you can cache the CDI lookup result rather than do the expensive lookup again and again

Reply:
Ok